src/pyclaw/solver.py

```diff
@@ -163,7 +178,8 @@ def __init__(self,riemann_solver=None,claw_package=None):
         self._is_set_up = False
         self._use_old_bc_sig = False
         self.accept_step = False
-        self.adjust_dt = False
+        self.get_dt = False
+        self.before_step = before_step
```

Comment:
I believe this should be initialized to None to avoid wasting time on needless function calls.